hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -283,16 +292,59 @@
       name: chip_usb_suspend
       desc: '''Verify that the USB device can detect the presence of VBUS from the USB host.
 
-            TBD.
+            - This test extends from `chip_usb_fs_df_tx_rx`, add below at the end of the sequence.
+            - Configure USB device to enter `Suspend` state and ensure `link_suspend` interrupt is
+              triggered.
+            - Test these 2 power modes.
+              - Normal sleep:
+                - Configure pwrmgr to enter normal sleep mode, then clocks are disable while powers
+                  are kept on.
+                - Resume the device through pinmux and check the `link_resume` interrupt.
+                - Ensure that previously enumerated information is kept.
+              - Deep sleep:
+                - Before entering deep sleep, store previously enumerated information in retention
+                  RAM. (optional)
+                - Configure pwrmgr to enter deep sleep mode, and powers are turned off.
+                - Resume the device through pinmux and check the `link_resume` interrupt.
```

Comment:
so i think i'm noticing what could be a bug on the `usbdev` side. @mdhayter / @vogelpi  let me know if you agree.
When we go into low power, the link state completely resets. Without issuing a reset event from the usb perspective, there may not be a way for the link state to return to active.  Without returning to active, our outgoing reference pulse information won't be correct, and the clock calibration will likely fail.

This is somewhat separate from the usb protocol handling which i think actually could still behave correctly once all the proper information is restored. We'll probably need to look at this area more before we can decide exactly what to do.

Reply:
> LGTM, pending Tim's comment below.

@sriyerg @weicaiyang 
i think maybe you guys should proceed with this PR first. Let me resolve this usb question separately.

Reply:
For spec compliance, it's actually not enough to merely have the capability to return to the active state. From the USB 2.0 spec, a device that goes into suspend must retain its device address and all configuration. A resume from suspend puts the device right back in the device state it left (i.e. wrt the USB Device State FSM).

Only a signaled reset or loss of VBUS would kick usbdev back to start.

Reply:
@vogelpi do you have an answer to Alex's question?

Reply:
Hey @a-will , @weicaiyang ,
you are right. The regular suspend will not alter the device's address and configuration. Upon resume, the device won't need to be reconfigured. See `usbdev_linkstate.sv` for details.

In the case of the the deep sleep feature discussed here, firmware will be in charge of backing up the config to AON retention SRAM and doing the restore. The only documentation on this I believe is this [one pager](https://docs.google.com/document/d/1ozcAiSU6N8k2L3e1k1v6anu6UoqEEedKPmUkzwW_2ng/edit#). In this https://github.com/lowRISC/opentitan/issues/3530#issuecomment-735306722 Mark Hayter further explained a bit on the timing and he also mentioned that it would probably be even okay if the link was reset in case the 23ms aren't sufficient to restore the state.

Reply:
@a-will it would be great if you can have a look at this area and let us know if you think we've put enough hooks in place to allow this to happen correctly.